### PR TITLE
[JS] Other (Video): Functions

### LINKED ISSUE #247: JS: Functions
# Functions
Functions in JavaScript

## Directory
```
Topics/5.Syntax/6.Functions
```

## Description
- What is scope?
- What is the lifetime of JavaScript variables?
- Basic function syntax.
- What are parameters in functions?
- Types of functions.
- Why use functions?

### PR DESCRIPTION
**Pull request by Sarfraz-droid**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #247  

<hr>

## Type of change 📝

- [x] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)

<hr>

## Domain of Contribution 📊

- [x] JavaScript

<hr>
 
## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

image

<hr>

### CODE REVIEW

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review.

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+Now a function looks like this..  
+Wait.. lets break it into different parts first
+
+```javascript
+function myFunction(p1, p2) {
+
+return p1 * p2; // The function returns the product of p1 and p2
+
+}
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:22]: Please indent and not use not-needed whitespace.
```suggestion
```javascript
function myFunction(p1, p2) {
    return p1 * p2; // The function returns the product of p1 and p2
}
```
```

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+if (true) {
+
+const a = 2;
+
+}
+
+console.log(a); //ReferenceError
+
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:43]:
```suggestion
``` javascript
if (true) {
    const a = 2;
}

console.log(a); //ReferenceError
```
```

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  
+Now lets go back to our function example:
+
+  
+```javascript
+function myFunction(p1, p2) {
+
+return p1 * p2; // The function returns the product of p1 and p2
+
+}
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:64]:
```suggestion
```javascript
function myFunction(p1, p2) {
    return p1 * p2; // The function returns the product of p1 and p2
}
```

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+```javascript
+	function sum(a,b)
+	{
+		c = a+b
+		console.log(c);
+	}
+
+	sum(10,12);
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:85]:
```suggestion
```javascript
function sum(a,b) {
	c = a+b
	console.log(c);
}

sum(10,12);
```
```

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+ - As the name suggests its anonymous and it doesn’t have a name to it.
+
+```javascript
+	const show = function(){
+		console.log("Anonymouse Function");
+	}
+	
+	show();
+	
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:102]:
```suggestion
```javascript
const show = function(){
	console.log("Anonymouse Function");
}
	
show();
```
```

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+ - This is useful when we have to call it many times and pass different
+       values to it
+
+```javascript
+function subtract(num1, num2)
+{
+	return num1 - num2;
+}
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:114]:
```suggestion
```javascript
function subtract(num1, num2) {
	return num1 - num2;
}
```
```

**Review by paramsiddharth — CHANGES_REQUESTED** [review]
Mostly format changes (critical to ASI). Please update the PR. Looks great to me!

**Comment by Sarfraz-droid** [comment]
Done @paramsiddharth

**Comment by Sarfraz-droid** [comment]
Done

**Review by s4-2 — APPROVED** [review]
Looks good to me:)

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi Please review.

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+Let’s see by an example
+
+``` javascript
+if (true) {
+    const a = 2;
+}
+
+console.log(a); //ReferenceError
+
+This time javascript throws an error. Why is it so?
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:36]:
```suggestion
```javascript
function myFunction(p1, p2) {
    return p1 * p2; // The function returns the product of p1 and p2
}

But before learning functions, lets learn about **scope**.

<hr>

### Scope

The scope is basically the availability of your code.  
Let’s see by an example

``` javascript
if (true) {
    const a = 2;
}

console.log(a); //ReferenceError
```

This time javascript throws an error. Why is it so?
```

You forgot to turn off the code-block.

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+ - As the name suggests its anonymous and it doesn’t have a name to it.
+
+```javascript
+const show = function(){
+	console.log("Anonymouse Function");
+}
+	
+show();
+
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:88]: Here too.

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+ - This is useful when we have to call it many times and pass different
+       values to it
+
+```javascript
+function subtract(num1, num2) {
+	return num1 - num2;
+}
+
+**Arrow Functions**
```
Comment (paramsiddharth) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:99]: Same here.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]
Please close the code-blocks properly. The rest looks great!

**Review by paramsiddharth — APPROVED** [review]
@Creatoon @dakshsethi Please review.

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
+
+
+## Javascript Functions
+
```
Comment (dakshsethi) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:4]: It should be `# JavaScript Functions`

**Review by dakshsethi — CHANGES_REQUESTED** [review]

**Comment by Sarfraz-droid** [comment]
Done @dakshsethi

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+
+# Javascript Functions
+
+<iframe  width="560"  height="315"  src="https://www.youtube.com/embed/LAl0vdiPVZQ"  title="YouTube video player"  frameborder="0"  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"  allowfullscreen></iframe>
+
+
+## Transcript
+
+A **JavaScript function** is a block of code designed to perform a particular task. Its basically dividing you program into different parts.
```
Comment (Creatoon) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:10]:
```suggestion
A **JavaScript function** is a block of code designed to perform a particular task. Its basically dividing your program into different functional components.
```

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+
+The scope is basically the availability of your code.  
+Let’s see by an example
+
+``` javascript
+if (true) {
+    const a = 2;
+}
+```
+console.log(a); //ReferenceError
```
Comment (Creatoon) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:34]: Write full reference error text

File: JavaScript/Topics/5.Syntax/6.Functions/Video.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+Because (**if**) block creates a scope for `a` and `a` can be used within that scope only.
+
+You are free to access the variable within its scope , but a variable cannot
+
```
Comment (Creatoon) [inline JavaScript/Topics/5.Syntax/6.Functions/Video.md:39]:
```suggestion
```
aren't these 2 lines are combined?

**Review by Creatoon — CHANGES_REQUESTED** [review]

**Comment by Creatoon** [comment]
Missing Points:

1)Behavior of this variable inside both type of function, normal one and arrow one

Just don't define THIS keyword fully, you have to talk about its behavior as its details will be covered in THIS topic.

**Comment by Sarfraz-droid** [comment]
@Creatoon Done

**Review by Creatoon — APPROVED** [review]

**Pull request merged**